### PR TITLE
Serve front-end assets from node_modules

### PR DESCRIPTION
**Pull request by he8us**
Hello,

While the development of #859 I noticed that the front end dependencies were copied in this repository or loaded from raw.githubusercontent.com. IMHO, that's a bad idea because:
- For the first case: more code to manage, and how to update?
- For the second case, how to know if the external library updates and introduce a breaking change?

So here's my proposal to be able to install the front-end dependencies through NPM

### CODE REVIEW

**Comment by he8us** [comment]
Apparently I introduced a vulnerable dependency, but snyk doesn't let me see it. 
And when I run `npm audit` no vulnerabilities are found.

**Comment by melvincarvalho** [comment]
Good catch & thanks for the PR!

I think it makes sense to remove dependencies on raw.githubusercontent.com

Re snyk, I tried to check but got : 

```
Unable to display this organisation
The organisation does not exist, or you do not have permissions to access it.
```

File: package.json [hunk 1]
```diff
@@ -62,11 +63,14 @@
...
     "mashlib": "^0.7.13",
     "mime-types": "^2.1.11",
+    "mkdirp": "^0.5.1",
+    "ncp": "^2.0.0",
     "negotiator": "^0.6.0",
     "node-fetch": "^2.1.2",
     "node-forge": "^0.7.4",
     "nodemailer": "^3.1.4",
     "oidc-op-express": "^0.0.3",
+    "owasp-password-strength-test": "^1.3.0",
```
[the remark this answers is not in the document]
Comment (he8us) [reply package.json]: It's already loaded by `default-views/account/register-form.hbs:70`.

Should I revert this line and put it back to github?

[the remark this answers is not in the document]
Comment (RubenVerborgh) [reply package.json]: Perfect, you can leave it there.

File: default-templates/new-account/index.html [hunk 2]
```diff
@@ -4,7 +4,7 @@
   <meta charset="utf-8">
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <title>{{#if name}}{{name}} &ndash; {{/if}}Solid Home</title>
-  <link rel="stylesheet" href="/common/css/bootstrap.min.css">
+  <link rel="stylesheet" href="/common/vendors/bootstrap/dist/css/bootstrap.min.css">
```
[the remark this answers is not in the document]
Comment (he8us) [reply default-templates/new-account/index.html]: Done

File: bin/fe-vendors.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      path.resolve(destinationPath, dependency),
+      (err) => {
+        if (err) {
+          return console.error(err)
+        }
+        console.log('Copied', dependency, '!')
+      }
+    )
+  })
+}
```
[the remark this answers is not in the document]
Comment (he8us) [reply bin/fe-vendors.js]: Done

**Comment by RubenVerborgh** [comment]
Was going to let @kjetilk and @megoth merge this, but I'll just put this in `develop` for now. We should just merge develop into the v4.2.0 branch soon.

**Comment by RubenVerborgh** [comment]
Regarding Snyk, I'm following up with @joshdcollins.

**Comment by RubenVerborgh** [comment]
I joined Snyk and can now see the problem:

```
Regular Expression Denial of Service (ReDoS)

    Vulnerable module: node-forge
    Introduced through: webid@0.3.10

Detailed paths and remediation

    Introduced through: solid-server@4.1.7 › webid@0.3.10 › node-forge@0.6.49
    Remediation: No remediation path available. 

Overview

node-forge is a native implementation of TLS (and various other cryptographic tools) in JavaScript.

Affected versions of this package are vulnerable to Regular Expression Denial of Service (ReDoS) attacks. This can cause an impact of about 10 seconds matching time for data 3K characters long.
```

**Comment by RubenVerborgh** [comment]
Fixed by d81182f3b7335839764a324c41e2163a79d34129.

**Pull request merged**